### ISSUE TITLE
upgrade to latest DAP (1.41.0)

### ISSUE THREAD

**Issue by akosyakov** [eclipse-theia/theia#8136]
I depends on https://github.com/eclipse-theia/theia/issues/7100

TODO:
- bump up a dependency version in `@theia/debug`
- align completion kinds in `DebugConsoleSession`
- ...

**Comment by tsmaeder** [comment]
Currently, we have `"vscode-debugprotocol": "^1.32.0"`, but the latest version is now 1.48.x. Highlights since 1.32.0 include

* Filters on exception breakpoints
* Step granularity (line, instruction, statement)
* Progress 
* Memory read & disassemble requests
* data breakpoints

I would welcome any of those features.

**Comment by EvilBeaver** [comment]
Exception breakpoint filters are very useful and it's sad they're not supported Theia. Requesting an update to 1.48+

**Comment by colin-grant-work** [comment]
We currently use version `1.51.0`, but additional UI functionality based on the new features has yet to be implemented.

**Comment by EvilBeaver** [comment]
@colin-grant-work

> We currently use version `1.51.0`, but additional UI functionality based on the new features has yet to be implemented.

Can you help me to find a place in UI's code where the new UI for exception breakpoints filters should be implemented?

**Comment by EvilBeaver** [comment]
@colin-grant-work yes, thanks :) I was asking where is this widget in the codebase? :) It's hard to find it for me, sorry. Can you tell me the modules where this widget is implemented/new UI should be implemented?

**Comment by colin-grant-work** [comment]
Ah, apologies for the misunderstanding, @EvilBeaver.

Breakpoints and breakpoint metadata are managed [here](https://github.com/eclipse-theia/theia/tree/master/packages/debug/src/browser/breakpoint), and the rendering is done [in this widget](https://github.com/eclipse-theia/theia/blob/master/packages/debug/src/browser/view/debug-breakpoints-widget.ts) on the basis of [this model's data](https://github.com/eclipse-theia/theia/blob/master/packages/debug/src/browser/view/debug-breakpoints-source.tsx).

**Comment by sgraband** [comment]
DAP upgraded to `@vscode/debugprotocol` ^1.51.0 (installed 1.68.0).
